include/glow/Backend/Backend.h

```diff
@@ -141,16 +160,13 @@ class BackendUsingGlowIR : public Backend {
 };
 
 /// Perform Backend Factory registration.
-#define REGISTER_GLOW_BACKEND_FACTORY(FactoryName, BackendClass,               \
-                                      SpecificBackendKind)                     \
-  class FactoryName : public BaseFactory<BackendKind, Backend> {               \
+#define REGISTER_GLOW_BACKEND_FACTORY(FactoryName, BackendClass, BackendName)  \
```

Comment:
Do we need to provide `BackendName` as an argument? Could we just ask `BackendClass::getBackendName()` for a name, assuming each backend would provide such a static function?

Or can we have use-cases, where e.g. the same backend is registered under custom or even multiple names?

Reply:
Yup `BackendClass::getBackendName()` this makes sense. I mentioned in the other comment that I plan to do so. But maybe I just add it here.

> Or can we have use-cases, where e.g. the same backend is registered under custom or even multiple names?

Not sure if we have this use case, but it could be easily done if needed.